### PR TITLE
Replace at-tip checks with utils functions from the latest sdk

### PR DESCRIPTION
**Pull request by septerr**
Fixes # .

### Motivation

Rosetta CLI checks to see if it is at tip at a few different places. These checks are not consistent.

For instance, before constructing transactions, the CLI checks if it is a tip based on (1) if the stored current block is within tip delay or (2) block returned by network/status is at tip (it is within tip delay or has sync status of true) and it is same as the stored current block.

Before broadcasting transactions, the CLI checks if it is at tip based only on if the stored current block is within tip delay.

These inconsistent checks were leading to the CLI getting stuck during transaction broadcast for chains that use sync status rather than tip delays.

### Solution

Use utility functions from the sdk's utils package to do at-tip checks consistently.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,9 +3,10 @@ module github.com/coinbase/rosetta-cli
 go 1.13
 
 require (
-	github.com/coinbase/rosetta-sdk-go v0.6.7
+	github.com/coinbase/rosetta-sdk-go v0.6.8
 	github.com/fatih/color v1.10.0
 	github.com/olekukonko/tablewriter v0.0.2-0.20190409134802-7e037d187b0c
+	github.com/segmentio/golines v0.0.0-20200824192126-7f30d3046793 // indirect
```
Comment (patrick-ogrady) [inline go.mod:9]: make sure to run go mod tidy

File: pkg/processor/broadcast_storage_helper.go [hunk 2]
```diff
@@ -49,9 +53,9 @@ func (h *BroadcastStorageHelper) AtTip(
 	ctx context.Context,
 	tipDelay int64,
 ) (bool, error) {
-	atTip, _, err := h.blockStorage.AtTip(ctx, tipDelay)
+	atTip, _, err := utils.CheckStorageTip(ctx, h.network, tipDelay, h.fetcher, h.blockStorage)
 	if err != nil {
-		return false, fmt.Errorf("%w: unable to determine if at tip", err)
+		return false, err
```
Comment (patrick-ogrady) [inline pkg/processor/broadcast_storage_helper.go:58]: nit: I'd keep the error wrapping that was here before

File: pkg/tester/data.go [hunk 3]
```diff
@@ -488,39 +488,18 @@ func (t *DataTester) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
-	// we are synced to tip.
-	if status.SyncStatus != nil &&
-		status.SyncStatus.Synced != nil &&
-		*status.SyncStatus.Synced &&
-		blockIdentifier.Index == status.CurrentBlockIdentifier.Index {
-		return true, blockIdentifier.Index, nil
-	}
-
-	return false, blockIdentifier.Index, nil
+	return atTip, blockIdentifier.Index, nil
```
Comment (patrick-ogrady) [inline pkg/tester/data.go:502]: we need to protect against `blockIdentifier` being `nil`: https://github.com/coinbase/rosetta-sdk-go/blob/d931a3a8f3eab4d0e1d264bd10fa30c1ebce94c0/utils/utils.go#L255

**Comment by patrick-ogrady** [comment]
You'll also want to modify this: https://github.com/coinbase/rosetta-cli/blob/23e6daf53528a2695b05e6a7f584d4f5e31c1023/cmd/root.go#L272

File: pkg/tester/data.go (hunk 3 above)
Comment (patrick-ogrady) [reply pkg/tester/data.go:502]: (only can happen in `CheckStorageTip` not `CheckNetworkTip`)

Comment (septerr) [reply pkg/tester/data.go:502]: good catch!

**Review by patrick-ogrady — APPROVED** [review]

**Pull request merged**